NPE in com.sixsq.slipstream.persistence.Run.validateQuota

BB2 fails at the moment.

```
Caused by: java.lang.NullPointerException
    at com.sixsq.slipstream.persistence.Run.validateQuota(Run.java:696)
    at com.sixsq.slipstream.run.RunListResource.createRun(RunListResource.java:175)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:606)
    at org.restlet.resource.ServerResource.doHandle(ServerResource.java:491)
```

--- a/jar-connector/src/main/java/com/sixsq/slipstream/configuration/Configuration.java
+++ b/jar-connector/src/main/java/com/sixsq/slipstream/configuration/Configuration.java
@@ -130,6 +130,11 @@ public class Configuration {
 				.getProperty(key));
 	}
 
+	public static boolean isQuotaEnabled() throws ValidationException {
+		return isEnabled(ServiceConfiguration.RequiredParameters
+			.SLIPSTREAM_QUOTA_ENABLE.getName());
+	}
+
 	/**
 	 * Return the singleton instance of a Configuration object. This method must
 	 * be synchronized to ensure that only one instance of this class is

--- a/jar-connector/src/main/java/com/sixsq/slipstream/connector/UserParametersFactoryBase.java
+++ b/jar-connector/src/main/java/com/sixsq/slipstream/connector/UserParametersFactoryBase.java
@@ -9,9 +9,9 @@ package com.sixsq.slipstream.connector;
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
- * 
+ *
  *      http://www.apache.org/licenses/LICENSE-2.0
- * 
+ *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
@@ -69,11 +69,11 @@ public abstract class UserParametersFactoryBase extends
 	}
 
 	protected void initBaseParameters() throws ValidationException {
-		if(isQuotaEnabled()) {
+		if(Configuration.isQuotaEnabled()) {
 			initQuotaParameter();
 		}
 	}
-		
+
 	protected void initQuotaParameter()	throws ValidationException {
 		putParameter(
 				Run.QUOTA_VM_PARAMETER_NAME,
@@ -81,15 +81,9 @@ public abstract class UserParametersFactoryBase extends
 						.getInstance()
 						.getParameters()
 						.getParameterValue(
-								constructKey(Run.QUOTA_VM_PARAMETER_NAME), 
+								constructKey(Run.QUOTA_VM_PARAMETER_NAME),
 								Run.QUOTA_VM_DEFAULT),
 				Run.QUOTA_VM_DESCRIPTION, true, true);
-	}
-
-	private static boolean isQuotaEnabled() throws ValidationException {
-		return Configuration
-				.isEnabled(ServiceConfiguration.RequiredParameters.SLIPSTREAM_QUOTA_ENABLE
-						.getName());
 	}
 
 	public Map<String, UserParameter> getParameters() {

--- a/jar-service/src/main/java/com/sixsq/slipstream/run/RunListResource.java
+++ b/jar-service/src/main/java/com/sixsq/slipstream/run/RunListResource.java
@@ -40,6 +40,7 @@ import org.restlet.resource.Get;
 import org.restlet.resource.Post;
 import org.restlet.resource.ResourceException;
 
+import com.sixsq.slipstream.configuration.Configuration;
 import com.sixsq.slipstream.connector.Connector;
 import com.sixsq.slipstream.connector.ConnectorFactory;
 import com.sixsq.slipstream.credentials.Credentials;
@@ -172,7 +173,9 @@ public class RunListResource extends BaseResource {
 
 			run = addCredentials(run);
 
-			run.validateQuota(Vm.usage(user.getName()));
+			if (Configuration.isQuotaEnabled()) {
+				run.validateQuota(Vm.usage(user.getName()));
+			}
 			
 			createRepositoryResource(run);
 

